### PR TITLE
Image viewer class

### PR DESCRIPTION
**Pull request by dave3d**

### CODE REVIEW

File: Code/IO/include/sitkImageViewer.h [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  /**@}*/
+
+  /** \brief Set/Get Debug flag.
+   *
+   * Turning on the Debug flag produces output showing the command string used to
+   * launch the viewing application.
+   * @{
+   */
+  static void SetGlobalDebug( const bool dbg );
+  static bool GetGlobalDebug();
```
Comment (blowekamp) [inline Code/IO/include/sitkImageViewer.h]: Is this a Global Default? or is there only one global state?

Please add On/Off methods as well.

File: Code/IO/include/sitkImageViewer.h [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  * to view images.
+  *
+  * These environment variables are only checked at SimpleITK's launch.
+  */
+class SITKIO_EXPORT ImageViewer
+{
+
+public:
+  ImageViewer();
+
```
Comment (blowekamp) [inline Code/IO/include/sitkImageViewer.h]: We should also have a `virtual std::String GetName() const` method like other SimpleITK objects.

File: Code/IO/include/sitkImageViewer.h [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  static int  m_GlobalViewerImageCount;
+  static bool m_GlobalAreDefaultsInitialized;
+
+  // global only
+  static std::vector<std::string> m_GlobalDefaultSearchPath;
+  static std::vector<std::string> m_GlobalDefaultExecutableNames;
+  static unsigned int             m_GlobalProcessDelay;
+
+  static std::string m_GlobalDefaultViewCommand;
+  static std::string m_GlobalDefaultFileExtension;
```
Comment (dave3d) [inline Code/IO/include/sitkImageViewer.h]: We need a Set/Get for GlobalDefaultFileExtension

**Comment by blowekamp** [comment]
Looks good! 👍 

Merging to next.

**Comment by blowekamp** [comment]
I missed this warning yesterday:
c:\d\vs15-32\simpleitk\code\io\src\sitkimageviewer.cxx(153): warning C4018: '<': signed/unsigned mismatch [C:\d\vs15-32\SimpleITK-build\SimpleITK-build\Code\IO\src\SimpleITKIO.vcxproj] [C:\d\vs15-32\SimpleITK-build\SimpleITK.vcxproj]

Can you please add another commit to this topic to address this warning? Thanks!

**Comment by blowekamp** [comment]
merging to next

**Pull request merged**